benchmark/simulate_hgts.py

```diff
@@ -0,0 +1,627 @@
+
+    Parameters
+    ----------
+    seq_donor: skbio.sequence.Sequence
+        Sequence object for donor genome
+    genes_donor: dictionary
+        a dictionary of genes (CDS) and their info, with the key being the
+        protein IDs and the value being a 5-element list including the
+        translated sequence, the raw nucleotide sequence, the start and end
+        positions in the donor genome
+    seq_recip: skbio.sequence.Sequence
+        Sequence object for recipient genome
+    genes_recip: dictionary
+        a dictionary of genes (CDS) and their info, with the key being the
+        protein IDs and the value being a 5-element list including the
+        translated sequence, the raw nucleotide sequence, the start and end
+        positions in the recipient genome
+    donor_genome_fp: string
+        file path to donor genome
+    recip_genome_fp: string
+        file path to recipient genome
+    output_dir: string
+        path to output directory
+    percentage_hgts: float
+        percentage of HGT genes to simulate (of total genes in recipient
+        genome)
+    orthologous_rep_prob: float
+        rate of orthologous replacement HGTs
+    log_f: file handler
+    threads: integer
+        number of threads to use
+
+    Returns
+    -------
+
+    """
+    # output dir for OrthoFinder results
+    proteomes_dir = join(output_dir, "proteomes")
+    if not exists(proteomes_dir):
+        makedirs(proteomes_dir)
+
+    # write donor and recipient genes to file
+    if verbose:
+        sys.stdout.write("Write donor and recipient genes to file.\n")
+    genes_donor_fp = join(
+        proteomes_dir, "%s_donor.faa" % basename(donor_genome_fp))
+    genes_recip_fp = join(
+        proteomes_dir, "%s_recip.faa" % basename(recip_genome_fp))
+    with open(genes_donor_fp, 'w') as genes_donor_f:
+        for gene in genes_donor:
+            genes_donor_f.write(">%s\n%s\n" % (gene, genes_donor[gene][0]))
+    with open(genes_recip_fp, 'w') as genes_recip_f:
+        for gene in genes_recip:
+            genes_recip_f.write(">%s\n%s\n" % (gene, genes_recip[gene][0]))
+    if verbose:
+        sys.stdout.write("\tDone.\n")
+
+    # simulate orthologous replacement
+    if orthologous_rep_prob > 0.0:
+        if verbose:
+            sys.stdout.write("\tSimulate orthologous replacement HGTs ...\n")
+        launch_orthofinder(proteomes_dir, threads, verbose=True)
+        date = time.strftime("%c").split()
+        day = date[2]
+        if int(day) < 10:
+            day = "0%s" % day
+        results_dir = join(
+            proteomes_dir, "Results_%s%s" % (date[1], day),
+            "WorkingDirectory")
+        species_ids, sequence_ids, orthologous_groups =\
+            parse_orthofinder(results_dir)
+        # no orthologs found, exit orthologous replacement simulation
+        if orthologous_groups:
+            seq_recip = simulate_orthologous_rep(genes_donor,
```

Comment:
`seq_recip` is reused here from the input parameters, is that expected? It also seems to be reused on line 503